python-client/giskard/models/model_explanation.py

```diff
@@ -1,20 +1,180 @@
+
+    # Select SHAP values, which explain model's predictions with the highest probability.
+    filtered_shap_values = list()
+
+    for sample_idx, sample_prediction_idx in enumerate(predictions):
+        prediction_explanation = shap_values[sample_prediction_idx][sample_idx]
+        filtered_shap_values.append(prediction_explanation)
+
+    return filtered_shap_values
+
+
+def _wandb_bar_plot(shap_explanations: shap.Explanation, feature_name: str) -> Any:
+    """Get wandb bar plot of shap values of the categorical feature."""
+    _FEATURE_COLUMN = "feature_values"
+    _SHAP_COLUMN = "shap_abs_values"
+
+    # Extract feature values and related shap explanations.
+    shap_values = shap_explanations[:, feature_name].values
+    feature_values = shap_explanations[:, feature_name].data
+
+    # We are interested in magnitude.
+    shap_abs_values = np.abs(shap_values)
+
+    # Calculate mean shap value per feature value.
+    df = pd.DataFrame(data={_FEATURE_COLUMN: feature_values, _SHAP_COLUMN: shap_abs_values})
+    shap_abs_means = pd.DataFrame(df.groupby(_FEATURE_COLUMN)[_SHAP_COLUMN].mean()).reset_index()
+
+    # Create bar plot.
+    table = wandb.Table(dataframe=shap_abs_means)
+    plot = wandb.plot.bar(
+        table, label=_FEATURE_COLUMN, value=_SHAP_COLUMN, title=f"Mean(Abs(SHAP)) of '{feature_name}' feature values"
+    )
+
+    return plot
+
+
+def _wandb_scatter_plot(shap_explanations: shap.Explanation, feature_name: str) -> Any:
+    """Get wandb scatter plot of shap values of the numerical feature."""
+    _FEATURE_COLUMN = "feature_values"
+    _SHAP_COLUMN = "shap_values"
+
+    # Extract feature values and related shap explanations.
+    shap_values = shap_explanations[:, feature_name].values
+    feature_values = shap_explanations[:, feature_name].data
+
+    # Create scatter plot.
+    df = pd.DataFrame(data={_FEATURE_COLUMN: feature_values, _SHAP_COLUMN: shap_values})
+    table = wandb.Table(dataframe=df)
+    plot = wandb.plot.scatter(
+        table, y=_FEATURE_COLUMN, x=_SHAP_COLUMN, title=f"'{feature_name}' feature values vs SHAP values"
+    )
+
+    return plot
+
+
+def _wandb_general_bar_plot(shap_explanations: shap.Explanation, feature_names: Iterable) -> Any:
+    """Get wandb bar plot of general shap mean values."""
+    _FEATURE_COLUMN = "feature"
+    _SHAP_COLUMN = "global_shap_mean"
+
+    # Calculate global shap means.
+    shap_general_means = list()
+
+    for feature_name in feature_names:
+        shap_general_means.append(np.abs(shap_explanations[:, feature_name].values).mean())
+
+    # Create bar plot.
+    df = pd.DataFrame(data={_FEATURE_COLUMN: feature_names, _SHAP_COLUMN: shap_general_means})
+    table = wandb.Table(dataframe=df)
+    plot = wandb.plot.bar(
+        table, label=_FEATURE_COLUMN, value=_SHAP_COLUMN, title="General Mean(Abs(SHAP)) across all features"
+    )
+
+    return plot
+
+
+def shap_to_wandb(model: BaseModel, dataset: Dataset, **kwargs) -> None:
```

Comment:
I think this should be split into two functions:
Similar to the way we do:
```py
scan_results = giskard.scan(model, dataset)
scan_results.to_wandb()
```
we should be able to do:
```py
explanation_results = giskard.explain_with_shap(model, dataset)
explanation_results.to_wandb()
```
so you can just name `explain_full` into `explain_with_shap` that should return a `dataclass` containing the `shap_values` (after removing `explainer`). This `dataclass` should support a `to_wandb()` method which is the rest of your `shap_to_wandb()` function.

Something like:
```py
@dataclass
class ShapResults:
  shap_values: np.array 

  def to_wandb(self):
    ...
```

Reply:
Gonna do this tomorrow.

Reply:
Done